python/mp/effects/bin.py

```diff
@@ -0,0 +1,99 @@
+import copy
+from mp import color
+from mp.effects import effect
+
+class Bin(effect.Effect):
+    """
+    This effect contains a list of other effects. Calling next() invokes the next() method
+    on all the contained effects. Rosary actually only contains this one effect, adding and
+    removing effects to and from it.
+
+    In other words, its turtles all the way down.
+    """
+
+    # Wish there were a better way than requiring this every time
+    dm = copy.deepcopy(effect.Effect.dm)
+
+    def __init__(self, bead_set, **kwargs):
+        super().__init__(name="bin", bead_set=bead_set, **kwargs)
+        self.effects = []
+        self.effect_id_num = 0
+
+    def effect(self, id):
+        """Return the Effect object of an active effect by specifying the Effect id."""
```

Comment:
This is the effect() accessor that was in Rosary. It still works the right way, but there's some indirection in order to access it: e.g.: rosary.bin.effect(0), or rosary.bin.effect(3).bin.effect(2) (nested effect).